File: C++/ArrayString/25_twoDifference.cpp
#include<bits/stdc++.h>
using namespace std;

bool findPair(int arr[], int size, int n){
    //code
    map<int,int>mp;
    for(int i=0;i<size;i++){

            int temp = n+arr[i];
            int temp2 = abs(n-arr[i]);
        if(mp.find(temp)!=mp.end() or mp.find(temp2)!=mp.end())
                return true;
                
        mp.insert({arr[i], i});
    }
    return false;
}

File: Java/ArrayString/25_twoDifference.java
public boolean findPair(int arr[], int n, int tar)
    {
        if(n == 1) return false;
        Arrays.sort(arr);
        int i = 0, j = 1;
        
        while(i<n && j<n){
            int diff = arr[j] - arr[i];
            
            if(i != j && diff == tar){
                return true;
            }
            else if(diff < tar){
                j++;    
            }
            else{
                i++;
            }
        }
        return false;
    }